pyActigraphy/io/cwa/cwa_class.py

```diff
@@ -1,237 +1,513 @@
+import os
 from datetime import datetime
-from struct import Struct
+from datetime import timedelta
+from tools import DataReader
+
+import pandas
+from numpy.linalg import norm
+
+class CWA(object):
+    __slots__ = ["__endianess", "__stream", "__meta",
+                 "__blocks",
+                 "__device",
+                 "__name", "__uuid", "__startTime",
+                 "__duration", "__period",
+                 "__Accel", "__Gyro", "__Mag",
+                 "__Ascale", "__Gscale", "__Mscale",
+                 "__Others"]
+    __hardware = {0x00: "AX3",
+                  0xff: "AX3",
+                  0x17: "AX3",
+                  0x64: "AX6"}
+    __axes = {1: "Axyz",
+              2: "Gxyz/Axyz",
+              3: "Gxyz/Axyz/Mxyz"
+              }
+
+    @classmethod
+    def IsValidFile(cls, filename):
+        """
+        Checks if given file is a valid cwa file.
+        Checks is performed by name (do it have .cwa)
+        and by leading bits, which must be 'MD'
+
+        Parameters
+        ----------
+        filename: str
+            path to an existing file
+
+        Returns
+        -------
+        bool
+            True if given file is valid cwa file
+            False otherwise
+
+        Raises
+        ------
+        TypeError
+            if passed parameters are of incompatible type
+        """
+
+        if not isinstance(filename, str):
+            raise TypeError("filename must be a string")
+        ext = os.path.splitext(filename)
+        if ext[1] != ".cwa":
+            return False
+
+        f = open(filename, "rb")
+        tag = f.read(2).decode("ASCII")
+        if tag == "MD":
+            f.close()
+            return True
+        f.close()
+        return False
+
+    def __init__(self, filename, ):
+        if not isinstance(filename, str):
```

Comment:
@nbeliy : why don't you use your IsValidFile method? Besides, you open and close the file twice (one for checking and one for reading). Can't you do it all at once?

Reply:
IsValidMethod is for facilitate the determination of file format -- just in master code, you can faslty check if given file is cwa file. It is not intended (but it can) be used in __init__